### PR TITLE
Add Iris classifier example

### PR DESCRIPTION
**Pull request by hugoabonizio**
Maybe feature scalers should be on an utils module of this library, I can send a PR. WDYT?

### CODE REVIEW

File: examples/iris_dataset/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    offset = batch_id * batch_size
+    x = x_train[offset...offset + batch_size]
+    target = y_train[offset...offset + batch_size]
+
+    output = net.forward(x)
+
+    loss = net.loss(output, target)
+
+    y_trues += target.shape[0].times.map { |i| target[i].to_a.index(target[i].to_a.max) }.to_a
+    y_preds += output.value.shape[0].times.map { |i| output.value[i].to_a.index(output[i].value.to_a.max) }.to_a
```
Comment (christopherzimmerman) [inline examples/iris_dataset/README.md:38]: I just merged a PR adding `arg(min|max)` that should make these types of checks a lot cleaner, you can now replace this with:

```
y_trues += target.argmax(1).to_a
y_preds += output.value.argmax(1).to_a
```

File: examples/iris_dataset/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  sigmoid_cross_entropy_loss
+end
+
+batch_size = 10
+
+10.times do |epoch|
+  y_trues = [] of Int32
+  y_preds = [] of Int32
+
+  (y_train.shape[0] // batch_size).times do |batch_id|
```
Comment (christopherzimmerman) [inline examples/iris_dataset/README.md:28]: Nice, I'm glad there is an example using batches now!

**Review by christopherzimmerman — COMMENTED** [review]
Looks awesome, just the one noted change.  ~Also, can you clarify what you mean by "feature scalars", haven't heard that term before.~

Whoops, I read scalar, not scaler.  Yes, feature scaling is definitely needed.  I think it might be better to build it as a gate supporting back-propagation rather than just as standalone functions.  It would be nice to use group normalization after convolutional layers for example.

If you're interested in working on this, I'd welcome any PRs.  If you're not sure how to convert the normalization methods to gates, if you have a PR just for the methods, I can add the backpropagation in (the backwards method is basically just returning a cache of the pre-normalized values I think).

**Comment by hugoabonizio** [comment]
I'm not sure if implementing group normalization as a gate in this scenario would work the same way. As I know, it is a common practice to [scale/normalize the features before feeding it to a neural network](https://stats.stackexchange.com/a/254815) (or SVMs, LRs, and other kinds of linear models), specially for tabular data like Iris.

I was think more of a sklearn's [`StandardScaler`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html), that has the whole dataset to compute the mean and standard deviation, instead of a batch normalization approach. I agree that a BN layer would be a great feature, specially for deep learning models, but I'm proposing a simpler method for using in MLP models.

What do you think @christopherzimmerman?

File: examples/iris_dataset/README.md (hunk 1 above)
Comment (hugoabonizio) [reply examples/iris_dataset/README.md:38]: Great addition!

**Comment by christopherzimmerman** [comment]
@hugoabonizio I think that would be a great addition.  I know that `StandardScaler` treats `NaN` as missing values and doesn't include them in normalization, so I might work on some iterators that skip missing values, so that the `mean`, `std` methods can have an option to ignore them.
